assets/quick-order-list.js

```diff
@@ -139,14 +139,60 @@ class QuickOrderList extends HTMLElement {
     const name = document.activeElement.getAttribute('name');
 
     const quantity = inputValue - cartQuantity;
-
-    if (cartQuantity > 0) {
+    this.cleanErrorMessageOnType(event);
+    if (inputValue == 0) {
       this.updateQuantity(index, inputValue, name, this.actions.update);
     } else {
-      this.updateQuantity(index, quantity, name, this.actions.add);
+      this.validateQuantity(event, name, index, inputValue, cartQuantity, quantity);
     }
   }
 
+  cleanErrorMessageOnType(event) {
+    event.target.addEventListener('keydown', () => {
+      event.target.setCustomValidity(' ');
+      event.target.reportValidity();
+    });
+  }
+  
+  validateQuantity(event, name, index, inputValue, cartQuantity, quantity) {
+    if (inputValue < event.target.dataset.min) {
```

Comment:
This is already this way on main. For context we set the `min` wrongly to 0 to avoid the `-` in the input from being disabled (natively) if the buyer is at the min of 5. This way the `-` can bring them back to 0